Trackers/iou_tracker/iou.py

```diff
@@ -0,0 +1,203 @@
+import numpy as np
+from ..track import Track
+from scipy.optimize import linear_sum_assignment as linear_assignment
+
+class iou_track(Track):
+    def __init__(self, id, bbox, hits, miss):
+        Track.__init__(self, id, bbox, hits, miss)
+
+class iou_tracker():
+    def __init__(self):
+        self.tracks = []
+        self.nextId = 0
+        self.max_lost=10
+        self.min_hits=3
+
+    def add_track(self, id, bbox):
+        """
+        Adding newly tracked objects  to track list.
+        
+        Args
+        id : new id to be assigned 
+        bbox : bounding box of the object
+        centroid : centroid of the object
+
+        Returns
+        None
+        """
+        
+        track = iou_track(id, bbox , 1, 0)
+        self.tracks.append(track)
+        self.nextId += 1
+        print("added id ",track.id, "succesfully")
+
+    def delete_track(self):
+        """
+        Deletes a centroid track object from tracks list if the max age is crossed.
+        
+        Args
+        None
+
+        Returns 
+        None
+        """
+
+        for track in self.tracks:
+            if(track.miss > self.max_lost):
+                print("deleted id ",track.id, "succesfully")
+                self.tracks.remove(track)
+    
+    def update(self, detections):
+        """
+        Returns the active list of centroid track objects
+
+        Args
+        detections : list of bbox of detected objects
+
+        Returns
+        tracks : list of objects of class Track
+        """
+
+        if(detections == []):
+            if(self.tracks == []):
+                return self.tracks
+            else:
+                # increase every objects miss by 1
+                for track in self.tracks:
+                    track.miss +=1
+                self.delete_track()
+                return self.tracks
+
+        matched, unmatched_dets, unmatched_trks = self.assign_detections_to_trackers(detections, iou_thrd = 0.3)  
+    
+        for i in matched:
```

Comment:
Instead of using `i`, it is better to unpack it in `trk`, 'det' or any relevant variables so they self document their purpose.